pallets/tasks/src/lib.rs

```diff
@@ -538,9 +538,16 @@ pub mod pallet {
 			task_id
 		}
 
-		fn finish_task(network: NetworkId, task_id: TaskId, result: Result<(), ErrorMsg>) {
+		pub(crate) fn finish_task(
+			network: NetworkId,
+			task_id: TaskId,
+			result: Result<(), ErrorMsg>,
+		) {
+			if TaskOutput::<T>::get(task_id).is_some() {
```

Comment:
But if a task is re-executed and passed then this is a problem bcz it wont let the task store the Ok() result

Reply:
It would already not store the Ok() result in the situation you described because there is already an early return in the submit_task_result extrinsic before we get to calling finish_task https://github.com/Analog-Labs/timechain/blob/22fe324d33f232d50e37bb048e2f3f6e36a13de1/pallets/tasks/src/lib.rs#L355

Reply:
I added this additional return for the edge case where finish_task is called twice as a consequence of the same submit_task_result call in which case IMO we should keep the first result (from process_events) instead of overwriting it with the user submitted result

Reply:
Thanks for pointing out

```suggestion
			if let Some(Ok(())) = TaskOutput::<T>::get(task_id) {
```
otherwise let the processing go.

I also think the previous early return is also probelmatic for restarting tasks but i think the check is there for chronicle members to not again and again submit the same result which belongs to same shard 

i would also like input of @dvc94ch in here.

Reply:
i mean for now PR looks good its just a normal flow for previous early return.

Reply:
yep contains_key() would be better

Reply:
opened this issue as a follow up and changed it to contains_key

Reply:
https://github.com/Analog-Labs/timechain/issues/1766